docs/howto/upgrade-cluster/aws.md

```diff
@@ -134,62 +165,86 @@ cluster is unused or that the maintenance is communicated ahead of time.
+      create only. We cannot choose a version ahead of the current k8s
       control plane version.
 
-   2. Add a new core node group (like `core-b`)
+      Make sure the cluster's jsonnet config's version field matches the
+      current version of the control plane and this version is **no more
+      than three minor versions** ahead of the node groups versions.
 
-      Rename (part 1/3) the config file's entry for the core node group
-      temporarily when running this command, either from `core-a` to `core-b` or
-      the other way around.
+   #### 5.2. Rename part 1: add a new core node group (like `core-b`)
+
+      Rename the `CLUSTER_NAME.jsonnet` config file's entry for the core node
+      group temporarily when running this command, either from `core-a` to `core-b` or
+      the other way around, then regenerate the `$CLUSTER_NAME.eksctl.yaml` file and
+      then create the new nodegroup.
+
+      ```bash
+      # re-generate an eksctl config file for use with eksctl
+      jsonnet $CLUSTER_NAME.jsonnet > $CLUSTER_NAME.eksctl.yaml
+      ```
 
       ```bash
+      # create a copy of the current nodegroup
       eksctl create nodegroup --config-file=$CLUSTER_NAME.eksctl.yaml --include="core-b"
       ```
 
-   3. Delete all old node groups (like `core-a,nb-*,dask-*`)
+   #### 5.3. Rename part 2: delete all old node groups (like `core-a,nb-*,dask-*`)
+
+      Rename the core node group again in the config to its previous name,
+      so the old node group can be deleted with the following command,
+      then regenerate the `$CLUSTER_NAME.eksctl.yaml` file and after delete
+      the original nodegroup.
 
-      Rename (part 2/3) the core node group again in the config to its previous
-      name, so the old node group can be deleted with the following command.
+      ```bash
+      # re-generate an eksctl config file for use with eksctl
+      jsonnet $CLUSTER_NAME.jsonnet > $CLUSTER_NAME.eksctl.yaml
+      ```
 
       ```bash
+      # delete the original nodegroup
       eksctl delete nodegroup --config-file=$CLUSTER_NAME.eksctl.yaml --include="core-a,nb-*,dask-*" --approve --drain=true
       ```
 
       Rename (part 3/3) the core node group one final time in the config to its
       new name, as that represents the state of the EKS cluster.
 
-   4. Re-create all non-core node groups (like `nb-*,dask-*`)
+   #### 5.4. Rename part 3: re-create all non-core node groups (like `nb-*,dask-*`)
 
       ```bash
       eksctl create nodegroup --config-file=$CLUSTER_NAME.eksctl.yaml --include="nb-*,dask-*"
       ```
 
-   5. Restore the k8s version in the config
+   #### 5.5. Restore the k8s version in the config
 
       We adjusted the k8s version in the config to influence the desired version
       of our created nodegroups. Let's restore it to what the k8s control plane
-      currently have.
+      currently have if not already.
 
-5. *Upgrad EKS add-ons (takes ~3*5s)*
+### 6. Upgrade EKS add-ons (takes ~3*5s)
```

Comment:
I'm not sure if the addons needs to be updated directly once per k8s minor version upgraded, I'm not sure. The safer call is to bundle this with "upgrade k8s control plane one minor version at the time (up to three times)" step.

I see downsides in doing it once per minor version as well. For example, one can get stuck debugging something that would have gone away if upgrading all the way first.

Reply:
👍🏼 I moved it under the control plane update step in https://github.com/2i2c-org/infrastructure/pull/3420/commits/f5ca3276a01c2af20cc32ad242811b569e4812dc